### PR TITLE
Correcting ScatterGatherHandler's internal handling of GatherEndpoint…

### PR DESCRIPTION
**Pull request by abdulbeard**
Correcting ScatterGatherHandler's internal handling of GatherEndpoint as per ISSUE 2261: https://github.com/spring-projects/spring-integration/issues/2261

### CODE REVIEW

**Comment by artembilan** [comment]
Cool! Thanks!

Would you mind adding some test improvement into the `ScatterGatherParserTests` calling the mentioned `stop()` and assert afterwards that `gatherEndpoint` property is stopped.

**Comment by abdulbeard** [comment]
gatherEndpoint is a private, and is created internally. In the case of `TestDistribution` and `TestAuction` this `gatherEndpoint` is null, which is why `.stop()` has no effect on it.
So a test can't be written for the state of `gatherEndpoint` without editing the `ScatterGatherHandler` class.
Or so it seems to me at least.

**Comment by artembilan** [comment]
That's not true:

https://github.com/spring-projects/spring-integration/blob/master/spring-integration-core/src/test/java/org/springframework/integration/scattergather/config/ScatterGatherParserTests.java#L76

So, that assert confirms that `gatherEndpoint` property is not null.
You can cast `scatterGather` variable to the `Lifecycle` and call its `stop()`:

```
Lifecycle gatherEndpoint = TestUtils.getPropertyValue(scatterGather, "gatherEndpoint", Lifecycle.class);
...
((Lifecycle) scatterGather).stop();
assertFalse(gatherEndpoint.isRunning());
```

Does it work for you?

**Comment by abdulbeard** [comment]
That works for me. PR updated 👍

File: spring-integration-core/src/test/java/org/springframework/integration/scattergather/config/ScatterGatherParserTests.java [hunk 1]
```diff
@@ -17,11 +17,7 @@
 package org.springframework.integration.scattergather.config;
 
 import static org.hamcrest.Matchers.instanceOf;
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
-import static org.junit.Assert.assertSame;
-import static org.junit.Assert.assertThat;
-import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.*;
```
Comment (artembilan) [inline spring-integration-core/src/test/java/org/springframework/integration/scattergather/config/ScatterGatherParserTests.java]: No asterisk imports, please.

I'll fix on merge tomorrow.

Thank you for the update!

**Review by artembilan — CHANGES_REQUESTED** [review]

File: spring-integration-core/src/test/java/org/springframework/integration/scattergather/config/ScatterGatherParserTests.java (hunk 1 above)
Comment (abdulbeard) [reply spring-integration-core/src/test/java/org/springframework/integration/scattergather/config/ScatterGatherParserTests.java]: IntelliJ got greedy and did this behind the scenes. Updated now 👍

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/f202b257752ce9687735eed5d053db9ff885ecb5 and cherry-picked to `4.3.x`.

@abdulbeard ,

Thank you for contribution.
Looking forward for more!

**Pull request closed without merging**